src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/repository/model/data/EventNotification.kt

```diff
@@ -35,4 +35,19 @@ class EventNotification(
   @Temporal(value = TemporalType.TIMESTAMP)
   @Column(name = "LAST_MODIFIED_DATETIME", nullable = false)
   val lastModifiedDateTime: LocalDateTime,
-)
+) {
+  override fun equals(other: Any?): Boolean {
+    if (other == null) return false
+    if (this === other) return true
+    if (other !is EventNotification) return false
+    if (eventId != other.eventId ||
+      hmppsId != other.hmppsId ||
+      eventType != other.eventType ||
+      url != other.url ||
+      lastModifiedDateTime != other.lastModifiedDateTime
+    ) {
+      return false
+    }
```

Comment:
Isn't line 43-49 doing the same check as line 41?

Reply:
no, 41 is testing whether they're actually the same object, wheras 43 - 49 are testing that the two objects have identical parameters, which we're testing for to determine equality